crates/connectors/src/connector_cpfp.rs

```diff
@@ -2,52 +2,63 @@
 //!
 //! Reference: <https://bitcoinops.org/en/topics/cpfp/>
 
-use bitcoin::{key::TapTweak, psbt::Input, Address, Network, ScriptBuf};
-use secp256k1::{schnorr, XOnlyPublicKey};
+use bitcoin::{psbt::Input, Address, Network, ScriptBuf};
+use bitcoin_bosd::Descriptor;
+use secp256k1::schnorr;
 use strata_bridge_primitives::scripts::taproot::finalize_input;
 
 /// Connector for adding outputs to a transaction for CPFP.
 ///
 /// It creates a taproot locking script with a public key that is assumed to be tweaked and expects
 /// a schnorr signature to finalize the input.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 pub struct ConnectorCpfp {
     /// The bitcoin network for which to generate output addresses.
     network: Network,
 
-    /// The public key used to create the child transaction in CPFP.
-    public_key: XOnlyPublicKey,
+    /// The descriptor used to create the child transaction in CPFP.
+    descriptor: Descriptor,
+
+    /// The address corresponding to the descriptor.
+    descriptor_address: Address,
 }
 
 impl ConnectorCpfp {
     /// Constructs a new CPFP connector.
-    pub const fn new(public_key: XOnlyPublicKey, network: Network) -> Self {
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the descriptor cannot be converted to an address.
+    pub fn new(descriptor: Descriptor, network: Network) -> Self {
+        // TODO Return error instead of unwrapping in next commit
```

Comment:
TODOs must come with an assignee.


```suggestion
        // TODO: (@sistemd) Return error instead of unwrapping in next commit
```